sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala

```diff
@@ -60,5 +70,23 @@ class ConsoleSinkProvider extends StreamSinkProvider with DataSourceRegister {
     new ConsoleSink(parameters)
   }
 
+  def createRelation(
+      sqlContext: SQLContext,
+      mode: SaveMode,
+      parameters: Map[String, String],
+      data: DataFrame): BaseRelation = {
+    // Number of rows to display, by default 20 rows
+    val numRowsToShow = parameters.get("numRows").map(_.toInt).getOrElse(20)
+
+    // Truncate the displayed data if it is too long, by default it is true
+    val isTruncated = parameters.get("truncate").map(_.toBoolean).getOrElse(true)
+
+    data.sparkSession.createDataFrame(
```

Comment:
You can just call `data.showInternal(numRowsToShow, isTruncated)`. This is a hack in ConsoleSink to avoid using a wrong planner. That's not a problem in the batch DataFrames.